console/src/main/scala/io/joern/console/cpgcreation/GoCpgGenerator.scala

```diff
@@ -8,22 +8,15 @@ import scala.util.Try
 /** Language frontend for Go code. Translates Go source code into Code Property Graphs.
   */
 case class GoCpgGenerator(config: FrontendConfig, rootPath: Path) extends CpgGenerator {
```

Comment:
Since this is `gosrc2cpg` as opposed to `go2cpg`, can you create a corresponding `GoSrcCpgGenerator.scala` instead? That way, the old `go2cpg` (commercial) does not need to be adapted.

Reply:
Should I rename the same class `GoCpgGenerator` to `GoSrcCpgGenerator` or create a new one? If I create new one, only one of them will be invoked when `go` source code is found.

Reply:
Yes, pelase create a new one because the old one is used by commercial code not visible in joern at the moment.

Reply:
@fabsx00 I tired to create new `GoSrcCpgGenerator` and tried to give first priority to `GoCpgGenerator` first and if it fails then invoke `GoSrcCpgGenerator`. But I couldn't come up with a cleaner way to do it out of the generator. Rather I changed the `GoCpgGenerator` to handle both the use case and check if `go2cpg` script is available, if yes then invoke `go2cpg` frontend. In its absence, it will invoke `gosrc2cpg` frontend.

Reply:
I have verified adding dummy `go2cpg.sh` locally and see if it gets invoked. And it works as expected.

Reply:
That's also a great solution, thank you.